src/core/components/layout/components/columns/styles.ts

```diff
@@ -135,7 +137,62 @@ const generateWidthCSS = (width: number | number[]) => {
 	return calculateWidth(width);
 };
 
-export const column = (width?: number | number[]) => {
+const generateSpanCSS = (span: number | number[]) => {
+	if (Array.isArray(span)) {
+		const breakpoints: Breakpoint[] = [
+			'mobile',
+			'tablet',
+			'desktop',
+			'leftCol',
+			'wide',
+		];
+
+		return span.reduce((styles, w, i) => {
+			return css`
+				${styles}
+				${from[breakpoints[i]]} {
+					${calculateSpan(w)};
+				}
+			`;
+		}, css``);
+	}
+
+	return calculateSpan(span);
+};
+
+const columnBreakpoints: Array<ColumnBreakpoint> = [
+	{ totalColumns: 4, rule: until.tablet },
+	{ totalColumns: 12, rule: between.tablet.and.leftCol },
+	{ totalColumns: 14, rule: between.leftCol.and.wide },
+	{ totalColumns: 16, rule: from.wide },
+];
+
+const calculateSpan = (span: number) => {
+	const columnBreakpointCss = columnBreakpoints.reduce(
+		(acc: string, cur: ColumnBreakpoint) => {
+			if (span === 0) return `display: none;`;
+			if (span >= cur.totalColumns) return `width: 100%; display: block;`;
+
+			const inferredWidth = span / cur.totalColumns;
+			const cssForBreakpoint = `{ width: calc((100% + ${space[5]}px) * ${inferredWidth} - ${space[5]}px); display: block; }\n`;
+
+			return acc + cur.rule + cssForBreakpoint;
+		},
+		'',
+	);
+
+	return css`
+		box-sizing: border-box;
+		flex: ${span ? '0 0 auto' : 1};
+		${columnBreakpointCss}
+	`;
+};
+
+export const column = (width?: number | number[], span?: number | number[]) => {
+	if (!width && (span || span === 0)) {
```

Comment:
I'd be inclined to split this function up and push all this logic into the component. What do you think about:

```ts
// always gets applied
export const column = css`
	box-sizing: border-box;
`;

// get applied when no width or span is specified
export const flexGrow = css`
	flex: 1;
`;

// width is specified
export const width = (value: number | number[]) => css`
	flex: 0 0 auto;
	${generateWidthCSS(value)};
`;

// span is specified
export const span = (value: number | number[]) => css`
	flex: 0 0 auto;
	${generateSpanCSS(value)};
`;
```

Reply:
That looks much cleaner! I've added it now.